src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs

```diff
@@ -46,76 +51,108 @@ private enum Direction
                 var uncompressedByteCount = bytes.Length;
                 var requestPayload = GetRequestPayload(bytes);
-                var request = BuildRequest(uri, connectionInfo, requestPayload);
 
-                // Check serializedData length < MaxPayloadSizeInBytes before sending
-                // if > silently drop the payload for now, ideally want to be able split the data
-                // into chunks to send
                 if (requestPayload.Data.Length > _configuration.CollectorMaxPayloadSizeInBytes)
                 {
-                    // Log that the payload is being dropped. 
+                    // Log that the payload is being dropped
                     Log.ErrorFormat("Request({0}): Dropped large payload: size: {1}, max_payload_size_bytes={2}", requestGuid, requestPayload.Data.Length, _configuration.CollectorMaxPayloadSizeInBytes);
                     _agentHealthReporter.ReportSupportabilityPayloadsDroppeDueToMaxPayloadSizeLimit(method);
                     return "{}"; // mimics what is sent by NoOpCollectorWire
                 }
 
-                var response = SendRequest(request, requestPayload.Data);
+                var httpClient = _httpClientFactory.CreateClient(connectionInfo.Proxy);
+
+                var request = new HttpRequestMessage
+                {
+                    RequestUri = uri
+                };
+
+
+                request.Headers.Add("User-Agent", $"NewRelic-DotNetAgent/{AgentInstallConfiguration.AgentVersion}");
+                request.Headers.Add("Timeout", _configuration.CollectorTimeout.ToString());
+
+                request.Headers.Add("Connection", "keep-alive");
+                request.Headers.Add("Keep-Alive", "true");
+                request.Headers.Add("ACCEPT-ENCODING", "gzip");
 
-                _agentHealthReporter.ReportSupportabilityDataUsage("Collector", method, uncompressedByteCount, new UTF8Encoding().GetBytes(response).Length);
+                var content = new ByteArrayContent(requestPayload.Data);
+                var encoding = (requestPayload.IsCompressed) ? requestPayload.CompressionType.ToLower() : "identity";
+                content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+                content.Headers.Add("Content-Encoding", encoding);
+                content.Headers.Add("Content-Length", requestPayload.Data.Length.ToString());
+
+                request.Content = content;
+
+                foreach (var header in _requestHeadersMap)
+                {
+                    request.Headers.Add(header.Key, header.Value);
+                }
+
+                if (_configuration.PutForDataSend)
+                {
+                    request.Method = HttpMethod.Put;
+                }
+                else
+                {
+                    request.Method = HttpMethod.Post;
+                }
+
+                var response = httpClient.SendAsync(request).GetAwaiter().GetResult();
```

Comment:
This is the line I mentioned in the meeting this morning - in certain circumstances - can lead to a deadlock and/or thread pool starvation. Unfortunately, at this point in my tenure at NR, I don't have enough context to know whether this bit of code ever *might* be subject to the potential cases where a deadlock would occur. So this may be a non-issue.

Some relevant articles: 
https://stackoverflow.com/questions/39007006/is-getawaiter-getresult-safe-for-general-use
https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html
https://www.nikouusitalo.com/blog/why-is-getawaiter-getresult-bad-in-c/

Reply:
I was aware of the deadlock that might happen, but I couldn't find a better way to force async call to sync call.

Reply:
@vuqtran88 Did you look into doing async-await and what would have to happen to make the `SendData` method async?

Reply:
Thanks for bringing this up, @tippmar-nr, as I don't think it's something we consider enough. I don't believe this functionality can deadlock -- this happens on a timer, not a UI thread or in a request context, and there should only ever be one Send happening at a time. And I agree with Vu that doing it property as an async call would touch too much code. I do think it's worth putting in a tech debt Story to revisit this later, though.